specification/azsadmin/resource-manager/storage/Microsoft.Storage.Admin/preview/2016-05-01/farms.json

```diff
@@ -0,0 +1,709 @@
+{
+    "swagger": "2.0",
+    "info": {
+        "version": "2015-12-01-preview",
+        "title": "StorageManagementClient",
+        "description": "The Admin Storage Management Client."
+    },
+    "host": "adminmanagement.local.azurestack.external",
+    "schemes": [
+        "https"
+    ],
+    "consumes": [
+        "application/json"
+    ],
+    "produces": [
+        "application/json"
+    ],
+    "paths": {
+        "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/Microsoft.Storage.Admin/farms/{farmId}": {
+            "put": {
+                "x-ms-examples": {
+                    "Create a new storage farm.": {
+                        "$ref": "./Examples/Farms/Create.json"
+                    }
+                },
+                "tags": [
+                    "Farms"
+                ],
+                "description": "Create a new storage farm.",
+                "operationId": "Farms_Create",
+                "parameters": [
+                    {
+                        "$ref": "storage.json#/parameters/SubscriptionIdParameter"
+                    },
+                    {
+                        "$ref": "storage.json#/parameters/ResourceGroupNameParameter"
+                    },
+                    {
+                        "$ref": "storage.json#/parameters/FarmIdParameter"
+                    },
+                    {
+                        "$ref": "storage.json#/parameters/ApiVersionParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/FarmParameters"

```

Comment:
Please define `FarmParameters` directly here as:

```
        "FarmParameters": {
            "name": "farmObject",
            "in": "body",
            "required": true,
            "schema": {
                "$ref": "#/definitions/FarmCreationProperties"
            },
            "description": "Farm object supplied in the body of the Put farm operation.",
        },
```

and remove `FarmParameters` from parameter section (/parameters/FarmParameters)

Reply:
Done, but not sure why this was needed. Could you explain?

Reply:
see the previous comment.